File: lec12.java
import java.util.*;
public class lec12{
    public static Scanner scn = new Scanner(System.in);

    public static HashMap<Integer,Integer> input(int n){
        HashMap<Integer, Integer> map = new HashMap<>();
        for (Integer i = 0; i < n; i++) {
            map.put(i, scn.nextInt());
        }
        return map;
    }

    public static void display(HashMap<Integer, Integer> map){
        for (Map.Entry<Integer, Integer> entries : map.entrySet()) {
            System.out.println(entries.getKey() + ": " + entries.getValue());
        }
    }

    public static <K, V> void displayGen(HashMap<K, V> map) {
        for (Map.Entry<K, V> entries : map.entrySet()) {
            System.out.println(entries.getKey() + ": " + entries.getValue());
        }
    }

    public static ArrayList<Integer> arrToMap(int n) {
        ArrayList<Integer> arr = new ArrayList<>();
        for (Integer i = 0; i < n; i++) {
            arr.add(scn.nextInt());
        }
        return arr;
    }

    public static HashMap<Integer, Integer> countFreq(ArrayList<Integer> arr) {
        HashMap<Integer, Integer> map = new HashMap<>();

        for (Integer i = 0; i < arr.size(); i++) {
            map.put(arr.get(i), map.getOrDefault(arr.get(i), 0) + 1);
        }
        return map;
    }

    public static void main(String[] args){
        displayGen(countFreq(arrToMap(scn.nextInt())));
    }
}

File: lec12.py
def input_map(n):
    map_data = {}

    for i in range(n):
        map_data[i] = int(input())

    return map_data


def display(map_data):
    for key, value in map_data.items():
        print(f"{key}: {value}")


def display_gen(map_data):
    for key, value in map_data.items():
        print(f"{key}: {value}")


def arr_to_map(n):
    arr = []

    for i in range(n):
        arr.append(int(input()))

    return arr


def count_freq(arr):
    map_data = {}

    for i in range(len(arr)):
        value = arr[i]

        if value in map_data:
            map_data[value] += 1
        else:
            map_data[value] = 1

    return map_data


# main
n = int(input())

display_gen(count_freq(arr_to_map(n)))